lib/auth/auth.go

```diff
@@ -945,9 +945,29 @@ func (a *Server) refreshRemoteClusters(ctx context.Context) {
 		return
 	}
 
+	// randomize the order to optimize for multiple auth servers running in parallel
+	rnd.Shuffle(len(remoteClusters), func(i, j int) {
+		remoteClusters[i], remoteClusters[j] = remoteClusters[j], remoteClusters[i]
+	})
+
+	// we want to limit the number of backend updates performed on each refresh to avoid overwhelming the backend.
+	updateLimit := defaults.RemoteClusterRefreshLimit
+	if dynamicLimit := (len(remoteClusters) / defaults.RemoteClusterRefreshBuckets) + 1; dynamicLimit > updateLimit {
+		// if the number of remote clusters is larger than RemoteClusterRefreshLimit × RemoteClusterRefreshBuckets,
+		// bump the limit to make sure all remote clusters will be updated within reasonable time.
+		updateLimit = dynamicLimit
```

Comment:
Could we cover this scenario with unit tests? Do you think it's worth the effort?

Reply:
Sure, that never hurts. Added tests in https://github.com/gravitational/teleport/pull/22455/commits/567a5c1d51d472a5da53b5e52b0d0193cdc23aae